File: TestDemo/TestDemo/ViewController.m
//
//  ViewController.m
//  TestDemo
//
//  Created by hongzhiqiang on 2018/10/22.
//  Copyright © 2018 hhh. All rights reserved.
//

#import "ViewController.h"
#import "LQWaveCircleProgressView.h"

@interface ViewController ()

@property (nonatomic, strong) LQWaveCircleProgressView *progressView;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.view.backgroundColor = [UIColor whiteColor];
    [self.progressView startLoadView];
    
    //模拟进度
    dispatch_after(dispatch_time(DISPATCH_TIME_NOW, (int64_t)(2 * NSEC_PER_SEC)), dispatch_get_main_queue(), ^{
        NSTimer *timer = [NSTimer scheduledTimerWithTimeInterval:2 target:self selector:@selector(onTimer) userInfo:nil repeats:YES];
        [timer fire];
    });
    
}


- (void)onTimer{
    if (self.progressView.progress >= 1.0) {
        return;
    }
    self.progressView.progress += 0.1;
}

- (LQWaveCircleProgressView *)progressView {
    if (!_progressView) {
        _progressView = [LQWaveCircleProgressView new];
        _progressView.frame = CGRectMake(0, 0, 200, 200);
        _progressView.center = self.view.center;
        [self.view addSubview:_progressView];
    }
    return _progressView;
}

@end

File: TestDemo_Swift/TestDemo_Swift/ViewController.swift
//
//  ViewController.swift
//  TestDemo_Swift
//
//  Created by hongzhiqiang on 2018/10/23.
//  Copyright © 2018 hhh. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        self.view.addSubview(self.progressView)
        self.progressView.startLoadView()
        
        //模拟进度
        DispatchQueue.main.asyncAfter(deadline: DispatchTime.now()+2) {
            let time = Timer.scheduledTimer(timeInterval: 2, target: self, selector: #selector(self.onTimer), userInfo: nil, repeats: true)
            time.fire()
         
        }
    }
    
    @objc func onTimer() {
        if self.progressView.progress > 1.0 {
            return
        }
        self.progressView.progress += 0.1
    }
    
    lazy var progressView: LQWaveCircleProgressView = {
        let progressView = LQWaveCircleProgressView()
        progressView.frame = CGRect(x: 0, y: 0, width: 200, height: 200)
        progressView.center = self.view.center
        return progressView
    }()

}

